src/ui/components/Explorer.js

```diff
@@ -49,44 +71,108 @@ export class Explorer extends React.Component<ExplorerProps, ExplorerState> {
       params: {...view.params()},
       showWeightConfig: false,
       recalculating: false,
+      anchorEl: null,
+      selectedIndex: 1,
+      name: null,
     };
   }
 
+  handleMenuClose = () => {
+    this.setState({
+      anchorEl: null,
+    });
+  };
+
   // Renders the dropdown that lets the user select a type
   renderFilterSelect() {
+    const plugins = this.state.view.plugins();
     const optionGroup = (declaration: PluginDeclaration) => {
       const header = (
-        <option
+        <MenuItem
           key={declaration.nodePrefix}
           value={declaration.nodePrefix}
           style={{fontWeight: "bold"}}
+          onClick={() =>
+            this.setState({
+              selectedIndex: 0,
+              anchorEl: null,
+              filter: declaration.nodePrefix,
+              name: declaration.name,
+            })
+          }
         >
           {declaration.name}
-        </option>
+        </MenuItem>
       );
-      const entries = declaration.nodeTypes.map((type) => (
-        <option key={type.prefix} value={type.prefix}>
+      const entries = declaration.nodeTypes.map((type, index) => (
+        <MenuItem
+          key={type.prefix}
+          value={type.prefix}
+          onClick={() =>
+            this.setState({
+              selectedIndex: index + 1,
+              anchorEl: null,
+              filter: type.prefix,
+              name: type.name,
+            })
+          }
+        >
           {"\u2003" + type.name}
-        </option>
+        </MenuItem>
       ));
       return [header, ...entries];
     };
     return (
-      <label>
-        <span style={{marginLeft: "5px"}}>Showing: </span>
-        <select
-          value={NullUtil.orElse(this.state.filter, "")}
-          onChange={(e) => {
-            const filter = e.target.value || null;
-            this.setState({filter});
-          }}
+      <>
+        <List component="div" aria-label="Device settings">
+          <ListItem
+            button
+            aria-haspopup="true"
+            aria-controls="filter-menu"
+            aria-label="filters"
+            onClick={(event) =>
+              this.setState({
+                anchorEl: event.currentTarget,
+              })
```

Comment:
Let's change the name back to "Filter" here
```suggestion
              this.setState({
                name: "Filter",
                anchorEl: event.currentTarget,
              })
```

Reply:
We could have `Filter: ${currentTarget}` to retain information

Reply:
Actually came up with something better @topocount. Check it

Reply:
How about we do "both, and"? I like your idea of having the filter name in the description (`Filter: ${currentTarget`), and I also like the idea of putting the filter in the table header.